data/plugins/skill/mining/mining.rb

```diff
@@ -58,23 +85,25 @@ def executeAction
       return
     end
 
+    # checks to see if your pulse is on 0 and restarts the animation & pulse.
+    if @counter == 0
+      mob.play_animation pickaxe.animation
+      @counter = pickaxe.pulses
+    end
+
     # check if we need to kick start things
     if @started
-      # count down and check if we can have a chance at some ore now
-      if @counter == 0
-        # TODO: calculate the chance that the player can actually get the rock
-
-        if mob.inventory.add(ore.id)
-          name = name_of(@ore.id).sub(/ ore$/, '').downcase
-
-          mob.send_message("You manage to mine some #{name}.")
-          skills.add_experience(Skill::MINING, ore.exp)
-          # TODO: expire the rock
+      if get_mining_rate pickaxe, level
+        if ore.respawn != 0
+          mob.inventory.add(ore_id.id)
+          if objId != 2491 then mob.send_message(granite_or_sandstone ? "You manage to quarry some #{name}." : "You manage to mine some #{name}.") end
+          mob.skill_set.add_experience(Skill::MINING, ore_id.experience)
+          if ore.respawn != -1 then
```

Comment:
`unless ore.respawn == -1`

Don't need the `then` unless it's a single-line statement